services/src/main/java/org/keycloak/services/resources/admin/ComponentResource.java

```diff
@@ -126,19 +129,24 @@ public Stream<ComponentRepresentation> getComponents(@QueryParam("parent") Strin
     @Consumes(MediaType.APPLICATION_JSON)
     public Response create(ComponentRepresentation rep) {
         auth.realm().requireManageRealm();
-        try {
-            ComponentModel model = RepresentationToModel.toModel(session, rep);
-            if (model.getParentId() == null) model.setParentId(realm.getId());
+        AtomicReference<Response> ref = new AtomicReference<>();
```

Comment:
Yeah, I thought about that as well - if we move forward with this PR I might do something there.

Reply:
I agree, but please as a separate PR

Reply:
On the other hand, it seems that the exception handling could remain part of the main block and not be performed inside the `runJobInTransaction` / `lockRealmsForModification` since the exceptions thrown are unchecked. That way we could apparently do without the `AtomicReference` completely.